chapters/annotations.tex

```diff
@@ -336,11 +336,17 @@ \section{Annotations for Code Generation}\label{annotations-for-code-generation}
 "Evaluate" "=" ( false | true )
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
-The annotation \lstinline!Evaluate! can occur in the component declaration, its type declaration, or a base-class of the type-declaration.  In the case of multiple conflicting annotations it is handled similarly to modifiers (e.g., an \lstinline!Evaluate! annotation on the component declaration takes precedence).  The annotation \lstinline!Evaluate! only has effect for a component declared with the prefix \lstinline!parameter!.
+The annotation \lstinline!Evaluate! can occur in the component declaration, its type declaration, or a base-class of the type-declaration.
+In the case of multiple conflicting annotations it is handled similarly to modifiers (e.g., an \lstinline!Evaluate! annotation on the component declaration takes precedence).
+The annotation \lstinline!Evaluate! is only allowed for parameters and constants.
 
-If \lstinline!Evaluate = true!, the model developer proposes to utilize the value for the symbolic processing. In that case, it is not possible to change the parameter value after symbolic pre-processing.
+Setting \lstinline!Evaluate = true! is only allowed for structural parameters and constants.
```

Comment:
```suggestion
Setting \lstinline!Evaluate = true! is only allowed for constants and parameters fulfilling the requirements to \willbedefined{structural parameter}s.
```

Reply:
For me the original formulation is a little too strong. There are no structural parameters in the language (unless we invent the structural keyword).

Reply:
The intention of this PR – to formalize _structural parameter_ – is to make this a well defined term in the language without making it something explicitly declared using a new keyword.  (I don't think a change that would require all Modelica code in the wild to be modified would ever be accepted.)

Reply:
I'm not sure how this will work, see other comments on Evaluate=true in libraries.

Reply:
Better now, with the relaxed rules for `Evaluate` and the added example?

Reply:
This conversation is very old and outdated in view of the last design meeting discussions.  Resolving.